src/basic/fd-util.c

```diff
@@ -864,3 +864,20 @@ int fd_get_diskseq(int fd, uint64_t *ret) {
 
         return 0;
 }
+
+int dir_fd_is_root(int dir_fd) {
+        _cleanup_close_ int pfd = -EBADF;
+        struct stat st, rst;
+
+        assert(dir_fd >= 0);
+
+        pfd = openat(dir_fd, "..", O_CLOEXEC|O_DIRECTORY|O_PATH);
```

Comment:
no need to open, just do:

```c
r = path_get_mnt_id_at(dir_fd, ".", &a);
if (r < 0)
        return r;
r = path_get_mnt_id_at(dir_fd, "..", &b);
if (r < 0)
        return r;

return a == b;
```

on modern kernels that's just two syscalls, which is great